pkgs/servers/matrix-synapse/default.nix

```
@@ -1,30 +1,39 @@
-{ stdenv, python2Packages, fetchurl, fetchFromGitHub }:
+{ stdenv, python2, fetchurl, fetchFromGitHub }:
+
 let
-  matrix-angular-sdk = python2Packages.buildPythonPackage rec {
-    name = "matrix-angular-sdk-${version}";
-    version = "0.6.8";
+  py = python2.override {
+    packageOverrides = self: super: {
+      matrix-angular-sdk = super.buildPythonPackage rec {
+        name = "matrix-angular-sdk-${version}";
+        version = "0.6.8";
 
-    src = fetchurl {
-      url = "mirror://pypi/m/matrix-angular-sdk/matrix-angular-sdk-${version}.tar.gz";
-      sha256 = "0gmx4y5kqqphnq3m7xk2vpzb0w2a4palicw7wfdr1q2schl9fhz2";
-    };
-  };
-  matrix-synapse-ldap3 = python2Packages.buildPythonPackage rec {
-    pname = "matrix-synapse-ldap3";
-    version = "0.1.3";
+        src = fetchurl {
+          url = "mirror://pypi/m/matrix-angular-sdk/matrix-angular-sdk-${version}.tar.gz";
+          sha256 = "0gmx4y5kqqphnq3m7xk2vpzb0w2a4palicw7wfdr1q2schl9fhz2";
+        };
+      };
 
-    src = fetchFromGitHub {
-      owner = "matrix-org";
-      repo = "matrix-synapse-ldap3";
-      rev = "v${version}";
-      sha256 = "0ss7ld3bpmqm8wcs64q1kb7vxlpmwk9lsgq0mh21a9izyfc7jb2l";
-    };
+      matrix-synapse-ldap3 = super.buildPythonPackage rec {
+        pname = "matrix-synapse-ldap3";
+        version = "0.1.3";
 
-    propagatedBuildInputs = with python2Packages; [ service-identity ldap3 twisted ];
+        src = fetchFromGitHub {
+          owner = "matrix-org";
+          repo = "matrix-synapse-ldap3";
+          rev = "v${version}";
+          sha256 = "0ss7ld3bpmqm8wcs64q1kb7vxlpmwk9lsgq0mh21a9izyfc7jb2l";
+        };
 
-    checkInputs = with python2Packages; [ ldaptor mock ];
+        propagatedBuildInputs = with super; [ service-identity ldap3 twisted ];
+
+        # ldaptor is not ready for py3 yet
+        doCheck = !super.isPy3k;
```

```suggestion
        doCheck = !self.isPy3k;
```
even though it doesn't matter in this case